samza-core/src/main/scala/org/apache/samza/checkpoint/CheckpointTool.scala

```diff
@@ -126,8 +127,9 @@ object CheckpointTool {
   }
 
   def apply(config: Config, offsets: TaskNameToCheckpointMap): CheckpointTool = {
-    val coordinatorStreamManager = new CoordinatorStreamManager(config, new MetricsRegistryMap())
-    new CheckpointTool(offsets, coordinatorStreamManager)
+    val metadataStore: CoordinatorStreamStore = new CoordinatorStreamStore(config, new MetricsRegistryMap())
+    metadataStore.init()
```

Comment:
Any reason to instantiate and init the metadata store here if it'll be just used once and closed in the `CheckpointTool#run()` method?

Reply:
Helps to build unit tests easily by injecting a mock MetadataStore as a dependency into the checkpointTool#run method.